trie-db/src/triedbmut.rs

```diff
@@ -826,7 +827,7 @@ where
 
 		match self.storage.destroy(handle) {
 			Stored::New(node) => {
-				let encoded_root = node.into_encoded::<_, C, H>(|child| self.commit_child(child) );
+				let encoded_root = node.into_encoded::<_, C, H>(|child, k| self.commit_child(child, k));
```

Comment:
Would it make sense to also apply the scramble to root (with empty partial key)? (a few line below)